hazelcast/src/main/java/com/hazelcast/internal/serialization/impl/GenericRecordQueryReader.java

```diff
@@ -246,11 +249,14 @@ private Object readIndexed(InternalGenericRecord record, String path, int index)
     }
 
     private Object readLeaf(InternalGenericRecord record, String path) {
-        if (!record.hasField(path)) {
```

Comment:
why removed?

Reply:
Due to the logic [here](https://github.com/hazelcast/hazelcast/blob/4690a0ff0ca99d290b5649f4f91f47b3172bf6b7/hazelcast/src/main/java/com/hazelcast/internal/serialization/impl/GenericRecordQueryReader.java#L104), in order not to call hasField twice, I needed to move `record.hasField` call outside of the `readLeaf` method.

Reply:
Also, see [the old version](https://github.com/hazelcast/hazelcast/blob/cc26879d04c9e1f5db80d0fb7500ab37875f32db/hazelcast/src/main/java/com/hazelcast/internal/serialization/impl/GenericRecordQueryReader.java#L101)